### ISSUE TITLE
Failing to retrieve federated users items using GraphQL query - works for non-federated users

### ISSUE THREAD

**Issue by kylekirkby** [aws-amplify/amplify-js#5736]
**Describe the bug**
Cannot retrieve a federated sign in users' items from Grapqhl API.
I've got 3 groups:
- free
- starter 
- business

I've got multiple test users that I've created for testing. The users I've created using the non-federated sign in are able to retrieve items that I've added to the DB via a Lambda function that's triggered via a custom GraphQL mutation. However, if I sign in with the Google Auth Provider, the GraphQL query `getTemplate` fails to return the users templates. Instead I get this error:

```json
{
  "data": {
    "getTemplate": null
  },
  "errors": [
    {
      "path": [
        "getTemplate"
      ],
      "data": null,
      "errorType": "Unauthorized",
      "errorInfo": null,
      "locations": [
        {
          "line": 2,
          "column": 3,
          "sourceName": null
        }
      ],
      "message": "Not Authorized to access getTemplate on type Query"
    }
  ]
}
```

My JS for querying the calling the template Query:

```js
 async loadTemplate() {
    await API.graphql(
      graphqlOperation(GetTemplate, { id: this.props.match.params.id })
    )
      .then((templateResult) => {
          this.setState({
            templateData: templateResult.data.getTemplate,
          });
         .....
      })
      .catch((error) => {
        console.log("FAILED TO LOAD TEMPLATE");
        this.setState({
          loadingTemplate: false,
        });
        console.log(error);
      });
  }
```

When I'm adding the templates via the Lambda function I'm adding the users sub ID provided in the event context. I've checked to make sure the federated users sub exists - it does. Template sub matches the sub for the federated user.

I'm guessing that the part that is failing is:
```js
await API.graphql(
      graphqlOperation(GetTemplate, { id: this.props.match.params.id })
    )
...
```

Schema for my Template type looks something like this (some properties removed):
```graphql
type Template
  @model
  @auth(rules: [{ allow: owner }, { allow: private, provider: iam }]) {
  id: ID!
  name: String!
  type: String!
}
```

**Expected behavior**
Be able to retrieve a federated users items from GraphQL query.

If you need any more details please let me know. I've got a huge stack together with Amplify at current :D 

Best regards,

Kyle

**_You can turn on the debug mode to provide more info for us by setting window.LOG_LEVEL = 'DEBUG'; in your app._**

**Comment by kylekirkby** [comment]
I believe this related to the auth directives I've set for the GraphQL schema not allowing federated sign in users to access/use graphql queries/mutations. Thoughts?

**Comment by kylekirkby** [comment]
So after much much much digging I found this thread/comment:
https://github.com/aws-amplify/amplify-cli/issues/1313#issuecomment-489227002

So updating my GraphQL schema to:
```graphql
type Template
  @model
  @auth(
    rules: [
      { allow: owner, identityField: "sub" }
      { allow: private, provider: iam }
    ]
  ) {
  id: ID!
  name: String!
  type: String!
}
```
Allows access to users that have signed in with Google and with Email/Password (Cognito Users).

The sub was being used to identify my templates that I added via my Lambda function (for usage limits) but the username was being used by default for checking ownership apparently.

Hope this helps someone!
